src/runtime/virtcontainers/clh.go

```diff
@@ -1217,22 +1217,28 @@ func (clh *cloudHypervisor) addVolume(volume types.Volume) error {
 		return err
 	}
 
-	if clh.config.VirtioFSCache == virtioFsCacheAlways {
-		clh.vmconfig.Fs = []chclient.FsConfig{
-			{
-				Tag:       volume.MountTag,
-				CacheSize: int64(clh.config.VirtioFSCacheSize << 20),
-				Socket:    vfsdSockPath,
-			},
-		}
-	} else {
-		clh.vmconfig.Fs = []chclient.FsConfig{
-			{
-				Tag:    volume.MountTag,
-				Socket: vfsdSockPath,
-			},
-		}
+	// disable DAX if VirtioFSCacheSize is 0
+	dax := clh.config.VirtioFSCacheSize != 0
+	cacheSize := int64(0)
+	// numQueues and queueSize are required, let's use the
+	// default values defined by cloud-hypervisor
+	numQueues := int32(1)
+	queueSize := int32(1024)
 
+	if clh.config.VirtioFSCache == virtioFsCacheAlways {
+		dax = true
+		cacheSize = int64(clh.config.VirtioFSCacheSize << 20)
+	}
```

Comment:
I think we got a conflict here, when we have both `VirtioFSCacheSize = 0` and `virtioFsCacheAlways`. This is also a bug in the original implementation, but it was `working` as `VirtioFSCacheSize = 0` will be omitted when the struct is serialized to json stream (and clh will use its default `VirtioFSCacheSize` value that is 8G.).

My understanding is `VirtioFSCacheSize` got higher priority than `VirtioFSCache` in our kata configuration file, 
given the way how we disable DAX in kata w/ qemu is merely setting `VirtioFSCacheSize = 0` (regardless of the cache mode). If my understanding is right, I'd suggest to remove this block, and always assign `cacheSize =: int64(clh.config.VirtioFSCacheSize << 20)` .